pkg/kubectl/apply/primitive_element.go

```diff
@@ -32,3 +36,27 @@ func (e PrimitiveElement) Merge(v Strategy) (Result, error) {
 }
 
 var _ Element = &PrimitiveElement{}
+
+// ConflictDetect implement for primitive element.
+// Merge "primitive element" is the base case of recurisve call of element merge
+// PrimitiveElement has conflict in following conditions :
+// 1. Recorded and remote values do not match when both of them exist
+// 2. Only one of recorded and remote value exists
+// Example: primitve element list has conflict if value exists in one of recorded/remote but missing in the rest one.
+func (e PrimitiveElement) ConflictDetect(v Strategy) (Result, error) {
+	if !checkPrimitive(e) {
+		return Result{}, NewConflictError().WithMessage("Conflict detected in key [%v], recorded value (%+v) and remote value (%+v)",
```

Comment:
typo 'Conflict'

If the `Type` field can be remove are methods `NewConflictError` and `WithMessage` needed, esp. ?  Rather than just:
```go
return Result{}, &ConflictError{Message: fmt.Sprintf("...", ...)}
```